.buildkite/pipeline.yml

```diff
@@ -1,10 +1,26 @@
-use_plugins: &use_plugins
+x_use_plugins: &use_plugins
     plugins:
       - automattic/bash-cache#v1.0.0: ~
       - automattic/git-s3-cache#add/vm-host-support:
           bucket: "a8c-repo-mirrors"
           repo: "wordpress-mobile/wordpress-ios/"
 
+# Common environment values for the `env` key.
+#
+# To add more, simply add another entry at the same indentation as `IMAGE_ID`.
+#
+# Usage in a step with only these values:
+#
+#   env: *common_env
+#
+# Usage in a step with other values:
+#
+#   env:
+#     <<: *common_env
+#     other_key: "other value"
+x_common_env: &common_env
+  IMAGE_ID: xcode-12.5.1
+
```

Comment:
What I don't really like about this approach of using a key before the anchor that has the exact same name as the anchor, is that it gives the incorrect impression, especially to someone new to YAML anchors, that this is a requirement and that an anchor always have to be defined on a key of the same name. It hasn't.

In fact the only reason why we usually put an arbitrary key like that before the definition of the anchor is to nicely isolate the node we want to define an anchor on, and to not define that anchor at the top level of the YAML, but instead "namespace" that definition to avoid polluting the rest of the YAML.

So to clarify all that and imho make it more readable, and because the only role is to define anchors at the top of the YAML but still not at top-level in the root node, I personally prefer using an array for the namespacing, with one item for each anchor to define. That also avoids the confusion for people new to anchors.

```yaml
# Common YAML nodes we will reuse multiple times in the YAML
common_params:
  # Common plugins to use with the `plugins:` key
 - &common_plugins
   - automattic/bash-cache#v1.0.0: ~
   - automattic/git-s3-cache#add/vm-host-support:
       bucket: "a8c-repo-mirrors"
       repo: "wordpress-mobile/wordpress-ios/"

  # Common environment values for the `env` key.
- &common_env
  IMAGE_ID: xcode-12.5.1
```

Reply:
I really, really like this. In particular, it's clean and easy to extend: one doesn't need extra documentation comments or policies to explain how to add new shared values.

Reply:
> What I don't really like about this approach of using a key before the anchor that has the exact same name as the anchor, is that it gives the incorrect impression, especially to someone new to YAML anchors, that this is a requirement and that an anchor always have to be defined on a key of the same name.

😅